sources/main.c

```diff
@@ -15,11 +15,15 @@
 static int	interactive(t_param *param)
 {
 	int	ret;
+	char	**tokens;
 
 	while (1)
 	{
 		free(param->cur_command);
 		param->cur_command = readline(TXT_INVITATION);
+		tokens = parse_command(param->cur_command);
```

Comment:
Let's keep all tokenization in the input_to_text_tree, and leave this part intact.